### ISSUE TITLE
Allow use of password as alternative to API key

### ISSUE THREAD

**Issue by lblackstone** [rackspace/rack#372]
It appears that an API key is currently the only supported credential type. Could this be expanded to alternatively use password auth?

For reference: https://developer.rackspace.com/docs/cloud-servers/v2/developer-guide/#authenticate-with-curl

**Comment by jrperritt** [comment]
We've talked about this quite a bit. Password authentication has thus far deliberately been excluded from `rack`, because we feel that API keys are less of a security risk for users and provide all the same functionality for API calls (from an authentication perspective). We aren't closing the door on this feature, but instead would like some clarification, since perhaps there are some use cases we haven't considered. Can you provide some reasons for wanting to authenticate with a password instead of an API key?

**Comment by lblackstone** [comment]
Fair enough. It's not a make or break requirement, but I was trying to convert some existing scripts that were using password auth. I can switch to using API keys instead.
